### ISSUE TITLE
False-positive unused_closure_parameter for `self` with backticks

### ISSUE THREAD

**Issue by werediver** [realm/SwiftLint#2006]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --path Sample.swift --no-cache --enable-all-rules
Linting Swift files at path Sample.swift
Linting 'Sample.swift' (1/1)
/Users/raman/Projects/tmp/Sample.swift:6:13: warning: Unused Closure Parameter Violation: Unused parameter "self" in a closure should be replaced with _. (unused_closure_parameter)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version 0.24.2
* Installed via Homebrew
* No configuration file

* No [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)
* Xcode version 9.2 (9C40b)
* A sample:

```swift
internal func f(_: (Any) -> Void) {}

internal final class Sample {

    internal func perform() {
        f { `self` in print(self) }
    }
}
```

I hoped that #1979 handles this, but no :(

**Comment by marcelofabri** [comment]
Given that [this is a compiler bug](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160118/007425.html), I'm not sure if/how we should handle this.

**Comment by werediver** [comment]
Wow. Good to know about this compiler bug.

Taking into account the compiler bug I don't think this should be handled in any other way 👌

**Comment by marcelofabri** [comment]
I'm closing this for now then. If anyone feels like we should handle this, feel free to comment here or open a new issue mentioning this one.
